packages/components/tests/integration/components/hds/alert/index-test.js

```diff
@@ -0,0 +1,214 @@
+    );
+    assert.dom('.flight-icon-alert-diamond').exists();
+  });
+
+  test('if an icon is declared, the icon should render in the component and override the default one', async function (assert) {
+    assert.expect(2);
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" @icon="clipboard-copy" />`
+    );
+    assert.dom('.flight-icon-clipboard-copy').exists();
+    await render(
+      hbs`<Hds::Alert @description="yo" @type="compact" @icon="clipboard-copy" />`
+    );
+    assert.dom('.flight-icon-clipboard-copy').exists();
+  });
+
+  test('it should display no icon when @icon is set to false', async function (assert) {
+    await render(
+      hbs`<Hds::Alert @type="inline" @title="yo" @icon={{false}} />`
+    );
+    assert.dom('.flight-icon').doesNotExist();
+  });
+
+  // TEXT (TITLE + DESCRIPTION)
+
+  test('it should render the title when the @title argument is provided', async function (assert) {
+    await render(hbs`<Hds::Alert @type="inline" @title="This is the title" />`);
+    assert.dom(this.element).hasText('This is the title');
+  });
+  test('it should render the description when the @description argument is provided', async function (assert) {
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="This is the description" />`
+    );
+    assert.dom(this.element).hasText('This is the description');
+  });
+  test('it should render both the title and the description when both the @title and @description arguments are provided', async function (assert) {
+    assert.expect(2);
+    await render(
+      hbs`<Hds::Alert @type="inline" @title="This is the title" @description="This is the description" />`
+    );
+    assert.dom('.hds-alert__title').hasText('This is the title');
+    assert.dom('.hds-alert__description').hasText('This is the description');
+  });
+  test('it should render rich HTML when the @description argument contains HTML tags', async function (assert) {
+    assert.expect(8);
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="Hello <strong>strong</strong> and <em>em</em> and <code>code</code> and <a href='#'>link</a>" />`
+    );
+    assert.dom('.hds-alert__description strong').exists().hasText('strong');
+    assert.dom('.hds-alert__description em').exists().hasText('em');
+    assert.dom('.hds-alert__description code').exists().hasText('code');
+    assert.dom('.hds-alert__description a').exists().hasText('link');
+  });
+
+  // ACTIONS
+
+  test('it should render an Hds::Button component yielded to the "actions" container', async function (assert) {
+    assert.expect(2);
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" id="test-alert" as |A|><A.Button @text="I am a button" @size="small" @color="secondary" /></Hds::Alert>`
+    );
+    assert
+      .dom(
+        '#test-alert .hds-alert__actions button.hds-button.hds-button--size-small.hds-button--color-secondary'
```

Comment:
[nitpick] (non-blocking) my experience has been that chaining selectors together like this makes future refactoring trickier. is there a way we could simplify these to more directly target what we're looking for in the DOM?

Reply:
I personally always leaned heavily on the use of `data-test-*` attributes while writing tests but I know there are other approaches. Not necessary to completely resolve to one approach for this PR but something I think is worth thinking about as part of our overall strategy.

Reply:
100% agree with you. It's super brittle. I have pushed a change that alleviates the problem (and is more semantic for what I was testing): https://github.com/hashicorp/design-system/pull/245/commits/fe7aea918b8436751454d826c35568757881936d Let me know if that works for you.

Thanks for noticing it.

Reply:
> I personally always leaned heavily on the use of data-test-*

Yes, but at the moment we've avoided scattering the components' code with `data-test-*` attributes everywhere, and I would take the risk and try to avoid doing it (until it lasts). Maybe we can get away with it :)
If later we'll see that there's no other way, we will systematically add these attributes and update the tests accordingly.

PS: if you want to have a broader discussion on this, maybe open an issue so we have tracking of the decisions made.